### PR TITLE
feat(cli): warn when the analyzed project is below the Svelte/Kit floor

### PR DESCRIPTION
**Pull request by oekazuma**
## Why

svelte-vitals's rules assume Svelte 5+ (runes) and SvelteKit 2+. Some rules — `correctness/stale-prop-derivation`, `correctness/prop-mutation` — key off `$props()`/`$state()`/`$derived()` syntax specifically and structurally cannot recognize the legacy (`export let` / `$:`) equivalent of the exact same bugs.

Until now, nothing told the user this. A project still on an older Svelte/SvelteKit gets a clean, high-scoring report — not because the code is fine, but because entire rule categories can never fire on it. That's a silent false sense of security, not a bug in any single rule.

## What changed

- New `checkVersionFloor()` in `packages/cli/src/providers/source/project.ts`: reads the analyzed project's declared `svelte`/`@sveltejs/kit` version range from `package.json` (same source `detectProject` already reads for the `@sveltejs/kit` presence check) and returns a warning string for each dependency below the floor (Svelte 5 / SvelteKit 2).
- Wired into `analyzeProject` (`packages/cli/src/index.ts`) — the warning(s) are folded into the existing `warnings` array (previously populated only from config-file warnings) and printed via the same `stderr`/`errorLog` mechanism, prefixed `svelte-vitals: `.
- **Advisory only**: never changes the exit code, never blocks analysis — the project still gets analyzed normally either way.
- Docs (`guides/cli.md`, ja included) gained a short "Supported Svelte/SvelteKit versions" section explaining this and linking the two affected rules.
- Version parsing is intentionally simple (leading integer of the semver range string, e.g. `^5.56.6` → `5`) rather than a full semver dependency — this is a soft warning, not a gate, so precision beyond "major version" isn't needed.

## Deliberately out of scope

This is the first of two planned changes. This PR only adds the warning. It does **not** change what `stale-prop-derivation`/`prop-mutation` actually detect — a Svelte-5-floor-compliant project can still have individual components on legacy (`export let`) syntax mid-migration, and those will keep being silently missed by those two rules regardless of this warning. That's tracked as a separate, deliberately-scoped follow-up (teaching `component-parse.ts`'s prop-name collection to also recognize `export let`, with mode-aware fix recommendations in each rule's message).

## Test plan
- [x] `pnpm build` / `pnpm typecheck` / `pnpm test` / `pnpm lint` / `pnpm check:publish`
- [x] `pnpm --filter docs build`
- [x] New `packages/cli/test/version-floor.test.ts` covers: below-floor svelte, below-floor kit, both below floor, both at/above floor, absent from package.json, unparsable range (never guesses), devDependencies fallback, missing/malformed package.json
- [x] Verified none of the existing `packages/cli/test/fixtures/*/package.json` fixtures are below the floor (no accidental new warnings breaking existing `warnings` assertions)
- No `packages/action/dist` change needed — release.yml now self-heals that after merge (#278)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added non-blocking warnings when analyzed projects declare Svelte versions below 5 or SvelteKit versions below 2.
  - Clarifies that analysis still runs, but rune-dependent findings may be incomplete for components using legacy syntax.
- **Documentation**
  - Added a “Supported Svelte/SvelteKit versions” section to the CLI guides, including the warning behavior and analysis limitations.
- **Tests**
  - Added automated coverage for version-floor warning logic, including edge cases for missing or malformed version ranges.

### CODE REVIEW

File: packages/cli/src/providers/source/project.ts [hunk 1]
```diff
@@ -17,6 +17,58 @@ export class ProjectError extends Error {
...
+ * correctness/stale-prop-derivation, correctness/prop-mutation) can't recognize the
+ * legacy (`export let`, `$:`) equivalent of the same bugs, so findings may be incomplete.
+ */
+const MIN_SVELTE_MAJOR = 5;
+const MIN_KIT_MAJOR = 2;
+
+/** The leading integer in a semver range specifier (`^5.56.6` → 5, `>=4.0.0` → 4); undefined for an empty/unparsable string (`*`, `latest`, `workspace:*`, …) — never guessed. */
+function leadingMajor(range: string | undefined): number | undefined {
+  const m = range ? /\d+/.exec(range) : null;
+  return m ? Number(m[0]) : undefined;
```
[the remark this answers is not in the document]
Comment (oekazuma) [reply packages/cli/src/providers/source/project.ts]: Confirmed and fixed in bdd4735 — anchored the match to the start of the (optionally operator-prefixed) string instead of searching for a bare digit anywhere in it. Verified file:../svelte-4 and github:sveltejs/svelte#v4 both now correctly resolve to undefined (no false warning), while genuine ranges (^5.56.6, >=4.0.0, <=5.0.0, bare 5.56.6, etc.) still resolve correctly. Added a regression test. (The OpenGrep child_process.exec note on the same line is a false positive — that's RegExp.prototype.exec, not child_process.)
